src/components/icon/icon.tsx

```diff
@@ -46,7 +46,7 @@ export class Icon {
     public size: IconSize;
 
     /**
-     * Name of the icon
+     * Name of the icon, or URL of image to display
```

Comment:
I can hear you sharpening your pitchforks from here… 😬 🙈

But seriously, if we add support for displaying images as a separate component, we'll have to add that _everywhere_. By just making the small changes made to this file*, we get image support everywhere that we already have support for icons.

*) The other changes are mostly adding tests and docs, and a little bit of refactoring.

Reply:
We could also add a new `src` attribute that would take precedence over `name`

Reply:
Yeah, true, that would separate concerns a _little_ bit at least…

But I think we would still have to touch every single component that uses `limel-icon` to implement support for it in them, right?

I guess it might be worth doing that… 🤔 It would be a simple change in each place at least. Smaller than having to add code for checking if we have a name or a url and switching between rendering a limel-icon or a "limel-image" or something.

I can check out a separate branch, make that change in limel-icon, and then make the corresponding change in one or two components using limel-icon, to see what it would look like 👍

Reply:
I've updated it to take a separate `src` attribute instead. It's definitely cleaner. But we don't have support "for free" in any of the components that are using limel-icon. I'll add support to limel-chip or something, so we can see what it takes.

Reply:
It's fairly simple, but still quite a lot of work…

Time for me to log off now though. Will continue tomorrow.

Reply:
> I actually liked using the `name` more than adding a new `src` prop or similar.

Yeah, I'm not sure which way I prefer myself yet. The component does become nicer when the properties are separate, but just being able to start using it anywhere that you can configure which icon to use, regardless of whether any layers in-between you and `limel-icon` have been updated, is a very powerful argument.

Ultimately, I don't think "the component is a little nicer this way" is a strong enough argument for separating it into two properties. What I'm more worried about is if we would be painting ourselves into a corner that we later come to regret if we use the `name` property for both.

Reply:
The consumer has to update their code regardless if they want to use an image, so changing `name` to `src` would be done at the same time. I guess it's only more work for us to make sure that the `src` attribute is compatible everywhere if we go down that road instead of relying on `name` for URLs

Reply:
> The consumer has to update their code regardless if they want to use an image

Well, yes and no. If a direct consumer wants to change their code so it uses an image, then yes, there's no difference. But there are two cases where there _is_ a difference:

1. If there is a consumer in-between, and that consumer hasn't updated their code, the end-consumer has to wait for them to do that.
2. If the icon is set in configuration, and the user can enter the name of the icon as free text, then they could immediately start using images without waiting for any code to be updated.

But you might be right. While we certainly have a lot of case 1, most of those instances are perhaps inside lime-elements anyway. And while case 2 remains true, that's not the primary target case for this feature anyway, since that is to allow displaying user avatars in place of icons, and that's programmatic use, not config. (Or at least it's not going to be config only.)

Alright, I'll continue on adapting our own uses of `limel-icon` then 👍 😅

Reply:
If you wanna go that path, to update things everywhere, I'd say adding a new feature in those places is more proper. In the sense that you either say 
```tsx
icon: {
	name: '',
	color: '',
	etc: …
}
```

or use the `Image` interface and add 
image: {
	src: '',
	alt: '',
	etc: …
}
```

Why should be integrated in limel-icon then?

Reply:
I mean the need wasn't really "_being able to replace any icon with any image_". The need was "We wanna show an image instead of icon on the chip."

I'm not sure if this implementation is correct.

Reply:
> I'm not sure if this implementation is correct. In many places, you might still want to keep the icon, even if there is an image. This is what we did for instance for the feed items:
> <img alt="image" width="187" src="https://private-user-images.githubusercontent.com/35954987/319162321-29bdb4c2-8352-418c-9343-3a38b2dacc5c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNDg0MzksIm5iZiI6MTcxMjE0ODEzOSwicGF0aCI6Ii8zNTk1NDk4Ny8zMTkxNjIzMjEtMjliZGI0YzItODM1Mi00MThjLTkzNDMtM2EzOGIyZGFjYzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDEyNDIxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMTA2NDRmNjYxOTMwNzFjMTMxNzA0OGQzNjJkMWIwOGMxOTY4ODNiN2E5ODU2MDNmOTgwNzVlNjUwNjNkOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qyGnYyOHRRgpRzXKbM6N3LbXI9yQz2zTai_zM3oIUvI">
> where the icon goes down since it still give meaning, and the image is replaced with the icon, visually.

Ok, sure, I see your point there. I would consider that a bit of a special case though. I'm sure we can still make those even if we have what's proposed here.

I'm a little unsure how to interpret you at this point though. Do you no longer see any usefulness from this feature then?

Reply:
Sorry if I sound harsh… I'm sure or I hope you understand that I am not criticizing your creative and great work and smart approach. I'm just trying to let you know what scares me if we go this way. This has benefits certainly, but I'm worried that it will kick us back very quickly and very hard.

We can still complete what you started, but by adding support for images in limel-chip and consequently in the consumers of  limel-chip. Maybe adding an `Image` interface that offers things that concern images, and using it in a few places is a better approach.

Reply:
I made this alternative PR. What do you think? https://github.com/Lundalogik/lime-elements/pull/2885